app/models/sample_type.rb

```diff
@@ -115,7 +117,13 @@ def can_delete?(user = User.current_user)
   end
 
   def can_view?(user = User.current_user, referring_sample = nil)
-    project_membership = (user && user.person && (user.person.projects & projects).any?)
+    return false if Seek::Config.project_single_page_advanced_enabled && !template_id.nil?
+
+    can_view_in_single_page?(user, referring_sample)
+  end
+
+  def can_view_in_single_page?(user = User.current_user, referring_sample = nil)
```

Comment:
As above, I feel the logic is the wrong way around here, as this method contains the default general behaviour, whereas can_view? contains logic specific to the single page view